Marlin/src/libs/MAX31865.cpp

```diff
@@ -485,16 +454,33 @@ uint8_t MAX31865::spixfer(uint8_t x) {
     return SPI.transfer(x);
 
   uint8_t reply = 0;
-  for (int i = 7; i >= 0; i--) {
-    reply <<= 1;
-    WRITE(_sclk, HIGH);
+  for (int i = 7; i >= 0; i--) {    
     WRITE(_mosi, x & (1 << i));
+    DELAY_NS(_spi_speed);
+    WRITE(_sclk, HIGH);
+    DELAY_NS(_spi_speed);
+    reply <<= 1;
+    if (READ(_miso)) reply |= 1;
     WRITE(_sclk, LOW);
-    if (READ(_miso))
-      reply |= 1;
+    DELAY_NS(_spi_speed);
   }
+  return reply;  
+}
+
+void MAX31865::softSpiBegin(uint8_t spi_speed) {
+  #ifdef MAX31865_DEBUG
+    SERIAL_ECHOLNPGM("Initializing MAX31865 Software SPI");
+  #endif
+
+  // Calculate the actual clock speed
+  uint8_t target_clock_speed = 10000000/(2^spi_speed);
```

Comment:
In C++ the `^` operator is XOR, not exponent! This should be `_BV32(spi_speed)` or `pow(2, spi_speed)`.